File: src/main/java/ru/nsu/shatalov/timetable/model/enums/RoomType.java
package ru.nsu.shatalov.timetable.model.enums;

public enum RoomType {
  Terminal("Terminal", 20, 0),
  Lecture("Lecture", 200, 1),
  General("General", 50, 2);

  RoomType(String label, int capacity, int integer) {
    this.label = label;
    this.capacity = capacity;
    this.integer = integer;
  }

  private final String label;

  private final int capacity;

  private final int integer;

  public String getLabel() {
    return label;
  }

  public int getCapacity() {
    return capacity;
  }

  public int toInt() {
    return integer;
  }
}

File: frontend/src/enums/RoomType.ts
enum RoomType {
  Terminal = 'Terminal',
  Lecture = 'Lecture',
  General = 'General',
}

// Additional properties and methods can be represented using utility objects
const RoomTypeInfo = {
  Terminal: { label: 'Terminal', capacity: 20, integer: 0 },
  Lecture: { label: 'Lecture', capacity: 200, integer: 1 },
  General: { label: 'General', capacity: 50, integer: 2 },
};

const getRoomTypeInfo = (roomType: RoomType) => {
  return RoomTypeInfo[roomType];
};

export { RoomType, RoomTypeInfo, getRoomTypeInfo };
